packages/@okta/vuepress-site/guides/sign-into-native-ui/handle-authentication-responses/iOS/handle-responses.md

```diff
@@ -0,0 +1,107 @@
+### Response handling
+
+To initiate a particular flow please make a call to one of the available functions in OktaAuthSdk.swift:
+
+```swift
+// Authentication
+public class func authenticate(with url: URL,
+                               username: String,
+                               password: String?,
+                               onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                               onError: @escaping (_ error: OktaError) -> Void)
+
+// Unlock account
+public class func unlockAccount(with url: URL,
+                                username: String,
+                                factorType: OktaRecoveryFactors,
+                                onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                                onError: @escaping (_ error: OktaError) -> Void)
+
+// Forgot password
+public class func recoverPassword(with url: URL,
+                                  username: String,
+                                  factorType: OktaRecoveryFactors,
+                                  onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                                  onError: @escaping (_ error: OktaError) -> Void)
+
+// Restore authentication
+public class func fetchStatus(with stateToken: String,
+                              using url: URL,
+                              onStatusChange: @escaping (_ newStatus: OktaAuthStatus) -> Void,
+                              onError: @escaping (_ error: OktaError) -> Void)
+```
+
+Please note that the closure `onStatusChange` returns `OktaAuthStatus` instance as a parameter. An instance of the `OktaAuthStatus` class represents the current status that is returned by the server. It's the developer's responsibilty to handle current status in order to proceed with the initiated flow. Check the status type by calling `status.statusType` and downcast to a concrete status class. Here is an example handler function:
+
+```swift
+func handleStatus(status: OktaAuthStatus) {
+
+    switch status.statusType {
+        case .success:
+            let successState: OktaAuthStatusSuccess = status as! OktaAuthStatusSuccess
+            handleSuccessStatus(successStatus: successStatus)
```

Comment:
Show the entire `handleSuccessStatus` method below in "Succes status handling" (see comment)